.ci/scripts/test_global_patch.py

```diff
@@ -18,52 +18,80 @@
 import subprocess
 import sys
 
-# test patching from command line
-err_code = subprocess.call(
-    [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]
-)
-assert not err_code
-from sklearn.svm import SVC, SVR
-
-assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
-assert not SVR.__module__.startswith("daal4py") and not SVR.__module__.startswith(
-    "sklearnex"
-)
-
-
-from sklearnex import patch_sklearn, unpatch_sklearn
-
-# test unpatching from command line
-err_code = subprocess.call([sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"])
-assert not err_code
-unpatch_sklearn()

```

Comment:
The state of the unpatch_sklearn() test was previously after a ```subprocess.call([sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"])``` not a 
```subprocess.call(
        [sys.executable, "-m", "sklearnex.glob", "patch_sklearn", "-a", "svc"]``` call, meaning the state of the test is different in the new implementation.  This is okay if this was what you desired


https://github.com/intel/scikit-learn-intelex/pull/2090#discussion_r1794733125

Reply:
Thanks, I had missed that detail entirely.

So I tried adding a new test for global state separately from just `SVC`, but the test actually ends up failing:
```python
@pytest.fixture
def patch_all_from_command_line():
    err_code = subprocess.call(
        [sys.executable, "-m", "sklearnex.glob", "patch_sklearn"]
    )
    assert err_code == os.EX_OK

    yield

    err_code = subprocess.call(
        [sys.executable, "-m", "sklearnex.glob", "unpatch_sklearn"]
    )
    assert err_code == os.EX_OK

def test_patching_all_from_command_line(patch_all_from_command_line):
    from sklearn.svm import SVC, SVR

    assert SVC.__module__.startswith("daal4py")
    assert SVC.__module__.startswith("sklearnex")
    assert SVR.__module__.startswith("daal4py")
    assert SVR.__module__.startswith("sklearnex")
```

should that be expected to succeed?

Reply:
I assume that its failing on
```assert SVC.__module__.startswith("daal4py")```. You can remove
```assert SVC.__module__.startswith("daal4py")``` and ```assert SVR.__module__.startswith("daal4py")```

Reply:
It also fails for `sklearnex`.

Reply:
I added it this new test as an extra commit here, let's see if it also fails in the CI.

Reply:
Could be a legit failure then, lets see

Reply:
Looks like it also fails in the CI jobs:
```
    def test_patching_all_from_command_line(patch_all_from_command_line):
        from sklearn.svm import SVC, SVR
    
>       assert SVC.__module__.startswith("daal4py") or SVC.__module__.startswith("sklearnex")
E       AssertionError: assert (False or False)
E        +  where False = <built-in method startswith of str object at 0x7f8679a04300>('daal4py')
E        +    where <built-in method startswith of str object at 0x7f8679a04300> = 'sklearn.svm._classes'.startswith
E        +      where 'sklearn.svm._classes' = <class 'sklearn.svm._classes.SVC'>.__module__
E        +  and   False = <built-in method startswith of str object at 0x7f8679a04300>('sklearnex')
E        +    where <built-in method startswith of str object at 0x7f8679a04300> = 'sklearn.svm._classes'.startswith
E        +      where 'sklearn.svm._classes' = <class 'sklearn.svm._classes.SVC'>.__module__

scripts/test_global_patch.py:83: AssertionError
```

Reply:
Yeah this is bad, i'll admit I don't know much about the ```glob``` use.  I wonder if sklearn has been pre-imported via sklearnex imports, and the change in glob not affecting it.  There is definitely something here.

Reply:
Yes, looks like that does indeed make a difference. If leaving it as the only test then it actually succeeds. Will try to find a workaround.

Reply:
I didn't manage to find any solution so in the end I just removed the test and left a comment in the file.